### PR TITLE
Rename def-env to def --env

### PR DESCRIPTION
**Pull request by arnau**
This PR changes all `def-env` to `def --env`. `def-env` was deprecated in 0.87.

There are  a few commits:

- Rename def-env to the new def --env: Changes the book in English
- (de) Rename def-env to the new def --env: Changes the book in German (I don't speak German)
- (zh-CN) Rename def-env to the new def --env: Changes the book in Chinese (I don't speak Chinese)
- ~Update command docs. Change after running `nu make_docs.nu`~
- ~Add docs for the experimental command category. **Please review this bit. I was not expecting `nu make_docs.nu` to generate these files.** It seems like it's due to a plugin I have installed which makes me question whether `make_docs.nu` should filter plugins out?.~

```
╭────────────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ version            │ 0.87.1                                                                                                                │
│ branch             │                                                                                                                       │
│ commit_hash        │ 3d631490bc17cfef2944072f9b7869cd9cac1e7b                                                                              │
│ build_os           │ macos-aarch64                                                                                                         │
│ build_target       │ aarch64-apple-darwin                                                                                                  │
│ rust_version       │ rustc 1.71.1 (eb26296b5 2023-08-03)                                                                                   │
│ rust_channel       │ 1.71.1-aarch64-apple-darwin                                                                                           │
│ cargo_version      │ cargo 1.71.1 (7f1d04c00 2023-07-29)                                                                                   │
│ build_time         │ 2023-11-25 17:11:03 +00:00                                                                                            │
│ build_rust_channel │ release                                                                                                               │
│ allocator          │ mimalloc                                                                                                              │
│ features           │ dataframe, default, extra, sqlite, trash, which, zip                                                                  │
│ installed_plugins  │ custom-value generate, custom-value generate2, custom-value update, from eml, from ics, from ini, from vcf, gstat,    │
│                    │ inc, query, query json, query web, query xml                                                                          │
╰────────────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯```

### CODE REVIEW

**Comment by amtoine** [comment]
@arnau 
could you revert the changes to the `commands/` directory so that we keep this PR focused on the `def-env` fix? :innocent: 

let's say the diff is... huge :wink:

**Comment by arnau** [comment]
reverted 👍

File: book/modules.md [hunk 1]
```diff
@@ -197,8 +197,7 @@ Submodules are modules inside modules. They are automatically created when you c
 
 The difference is that `export module some-module` _only_ adds the module as a submodule, while `export use some-module` _also_ re-exports the submodule's definitions. Since definitions of submodules are available when importing from a module, `export use some-module` is typically redundant, unless you want to re-export its definitions without the namespace prefix.
 
-> **Note**
-> `module` without `export` defines only a local module, it does not export a submodule.
+> **Note** > `module` without `export` defines only a local module, it does not export a submodule.
```
Comment (amtoine) [inline book/modules.md:200]: this looks like it will break the note, right? :thinking:

File: book/modules.md [hunk 2]
```diff
@@ -515,5 +518,4 @@ It can be one of the following:
 
 - Hides all of the module's exports, without the prefix
 
-> **Note**
-> `hide` is not a supported keyword at the root of the module (unlike `def` etc.)
+> **Note** > `hide` is not a supported keyword at the root of the module (unlike `def` etc.)
```
Comment (amtoine) [inline book/modules.md:521]: same comment about this note

File: de/book/hooks.md
Comment (amtoine) [inline de/book/hooks.md:1]: brr i don't like when some editors add `\r` or `\n` to all the lines...
it breaks the changes :sob: :laughing:

File: book/modules.md (hunk 1 above)
Comment (arnau) [reply book/modules.md:200]: definitely. fixing it now.

**Review by amtoine — CHANGES_REQUESTED** [review]
thanks for going through these, just a few remarks :wink:

File: de/book/hooks.md
Comment (arnau) [reply de/book/hooks.md:1]: agreed. let me see if I can redo the change in a different editor. Helix seems to be overdoing it.

File: book/modules.md (hunk 1 above)
Comment (arnau) [reply book/modules.md:200]: Ok, something is not right here. I now have triple-checked and this is not the editor's fault but a git hook messing with Markdown.

**Comment by arnau** [comment]
@amtoine could you have a fresh look please? I've done it now from scratch without git hooks. All markdown issues are gone as far as I can tell.

**Review by amtoine — APPROVED** [review]
oooh yeah, looks much better, thanks a lot :pray: 

hopefully the last `def-env` are gone for good :relieved:

**Pull request merged**